doc/glossary.rst

```diff
@@ -0,0 +1,1216 @@
+        using this sense as a parameter name.
+
+    ``n_components``
+        TODO
+
+    ``n_jobs``
+        This is used to specify how many concurrent processes/threads should be
+        used for parallelized routines.  Scikit-learn uses one processor for
+        its processing by default, although it also makes use of NumPy, which
+        may be configured to use a threaded numerical processor library (like
+        MKL).
+
+        ``n_jobs`` is an int, specifying the maximum number of concurrently
+        running jobs.  If set to -1, all CPUs are used. If 1 is given, no
+        parallel computing code is used at all.  For n_jobs below -1, (n_cpus +
+        1 + n_jobs) are used. Thus for n_jobs = -2, all CPUs but one are used.
+
+        The use of ``n_jobs``-based parallelism in estimators varies:
+
+        * Most often parallelism happens in :term:`fitting <fit>`, but
+          sometimes parallelism happens in prediction (e.g. in random forests).
+        * Some parallelism uses a multi-threading backend by default, some
+          a multi-processing backend.  It is possible to override the default
+          backend by using :func:`sklearn.externals.joblib.parallel.parallel_backend`.
+        * Whether parallel processing is helpful at improving runtime depends
+          on many factors, and it's usually a good idea to experiment rather
+          than assuming that increasing the number of jobs is always a good
+          thing.  *It can be highly detrimental to performance to run multiple
+          copies of some estimators or functions in parallel.*
+
+        Nested uses of ``n_jobs``-based parallelism with the same backend will
+        result in an exception.
+        So ``GridSearchCV(OneVsRestClassifier(SVC(), n_jobs=2), n_jobs=2)``
+        won't work.
+
+        When ``n_jobs`` is not 1, the estimator being parallelized must be
+        picklable.  This means, for instance, that lambdas cannot be used
+        as estimator parameters.
+
+    ``random_state``
+        Whenever randomization is part of a Scikit-learn algorithm, a
+        ``random_state`` parameter may be provided to control the random number
+        generator used.  Note that the mere presence of ``random_state`` doesn't
+        mean that randomization is always used, as it may be dependent on
+        another parameter, e.g. ``shuffle``, being set.
+
+        ``random_state``'s value may be:
+
+            None (default)
+                Use the global random state from :mod:`numpy.random`.
+
+            An integer
+                Use a new random number generator seeded by the given integer.
+                To make a randomized algorithm deterministic (i.e. running it
+                multiple times will produce the same result), an arbitrary
+                integer ``random_state`` can be used. However, it may be
+                worthwhile checking that your results are stable across a
```

Comment:
You could mention that 42 is used commonly as a joke? People have asked me why 42 in the past...